.NET/Microsoft.Recognizers.Text.DateTime/Utilities/TasksModeSetHandler.cs

```diff
@@ -0,0 +1,325 @@
+                }
+
+                var futureValue = refDate.AddDays(7);
+
+                // value = "09-04-2022 19:30" to extract only date substring from value used value[0:10].
+                if (DateTimeFormatUtil.FormatDate(futureValue).Equals(value.Substring(TasksModeConstants.IntDateStartIdx, TasksModeConstants.IntDateEndIdx)) && result.Timex.StartsWith(TasksModeConstants.FuzzyYearAndWeek))
+                {
+                    if (result.Timex.Contains(Constants.TimeTimexPrefix))
+                    {
+                        if (DateTimeFormatUtil.FormatTime(refDate).CompareTo(value.Substring(11)) <= 0)
+                        {
+                            value = JoinDateWithValue(refDate, value.Substring(11));
+                        }
+                    }
+                    else
+                    {
+                        value = DateTimeFormatUtil.FormatDate(refDate);
+                    }
+                }
+
+                result.FutureResolution = new Dictionary<string, string>
+                {
+                    { resKey, (string)value },
+                };
+
+                result.PastResolution = new Dictionary<string, string>
+                {
+                    { resKey, (string)value },
+                };
+            }
+            else if (result.Timex.StartsWith(Constants.TimeTimexPrefix))
+            {
+                var timexRes = TimexResolver.Resolve(new[] { result.Timex }, refDate);
+
+                string value = GetStartValue(timexRes);
+                if (value == null)
+                {
+                    value = timexRes.Values[0].Value;
+                }
+
+                DateObject resDate = refDate;
+                if (DateTimeFormatUtil.FormatTime(resDate).CompareTo(value) > 0)
+                {
+                    resDate = resDate.AddDays(1);
+                }
+
+                result.FutureResolution = new Dictionary<string, string>
+                {
+                    { TimeTypeConstants.DATETIME, JoinDateWithValue(resDate, (string)value) },
+                };
+
+                result.PastResolution = new Dictionary<string, string>
+                {
+                    { TimeTypeConstants.DATETIME, JoinDateWithValue(resDate, (string)value) },
+                };
+            }
+            else
+            {
+                result.FutureResolution = new Dictionary<string, string>
+                {
+                    { TimeTypeConstants.SET, (string)result.FutureValue },
+                };
+
+                result.PastResolution = new Dictionary<string, string>
+                {
+                    { TimeTypeConstants.SET, (string)result.PastValue },
+                };
+
+            }
+
+            return result;
+        }
+
+        public static string ReplaceValueInTextWithFutTerm(string text, string value, List<string> thisTermList)
+        {
+            value = value.Trim();
+            string thisTerm = thisTermList[0];
```

Comment:
Again, don't just use the first value in the list. If it has more than one, you should check more then one.

Reply:
added comment as per our discussion.